### PR TITLE
Auto: implement issue #718

### LINKED ISSUE #718: Add retry-scheduling columns to WebhookDeliveryModel and migration
Subtask of #496

Add `next_retry_at` (DateTime, nullable, index), `last_attempt_at` (DateTime, nullable), and `error_message` (Text, nullable) columns to `WebhookDeliveryModel` in `src/db/models/webhook.py`. Generate an alembic migration: spin up alembic_dev DB, run `alembic revision --autogenerate -m 'add webhook delivery retry columns'`, then review and fix: ensure JSON→JSONB, DateTime→DateTime(timezone=True), and `next_retry_at` has a partial index `ix_delivery_next_retry` on `next_retry_at WHERE next_retry_at IS NOT NULL`. Verify `alembic upgrade head && alembic downgrade -1 && alembic upgrade head` exits 0.

Dev-Plan: docs/dev-plan/50-automation/0718-add-retry-scheduling-columns-to-webhookdeliverymodel-and-mig.md

### PR DESCRIPTION
**Pull request by yihua-zhuo**
Closes #718

Plan: `.plans/issue-718.md`

This PR was auto-implemented by Claude Code in three stages:

1. **Plan** — Claude wrote a detailed implementation plan to `.plans/issue-718.md`.
2. **Implement** — Claude executed the plan, ran `ruff check src/` + the unit and integration test suites, and iterated until all three were green.
3. **Self-review** — Claude reviewed its own diff against `codereview.md` (the project review checklist) and fixed any rule violations found, re-running the test suite if any changes were made.

PR-time CI (`Dev Agent CI`) re-verifies independently before this PR is mergeable.

Please review before merging.

## New Features
* Webhook delivery tracking enhanced with retry scheduling information and error logging:
  * Track next scheduled retry attempt time
  * Record timestamp of last delivery attempt
  * Capture detailed error messages from failed deliveries

[Review Change Stack](https://app.coderabbit.ai/change-stack/yihua-zhuo/agent-job/pull/744?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by yihua-zhuo** [comment]
## Claude Code Review (inline)

**Status:** `pass`
**Inline comments posted:** 2

**Summary:** Reviewed 4 file(s) between origin/master and origin/auto/issue-718.

### Issues without a usable file/line
- `alembic/versions/a940f167ad16_add_webhook_delivery_retry_columns.py`: Migration is clean — columns types match the model, partial index condition is correct, downgrade is properly ordered, and no violations of migration guidelines.
- `src/db/models/webhook.py`: Model correctly adds three nullable fields and extends to_dict() with timezone-aware ISO serialisation; partial index in __table_args__ matches the migration's postgresql_where clause.
- `tests/unit/test_webhook_model.py`: Test suite covers all new fields — presence check, to_dict serialisation with values, and null edge cases; existing tests remain intact and updated test_columns_present includes the new fields.

File: .plans/issue-718.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+## Goal
+
+Add three retry-scheduling columns (`next_retry_at`, `last_attempt_at`, `error_message`) to `WebhookDeliveryModel` and generate a corrected Alembic migration that includes a partial index on `next_retry_at`. This is prerequisite groundwork for the background retry scheduler (#721).
+
+## Source Contract
+
+Dev-plan target: `/home/runner/work/agent-job/agent-job/docs/dev-plan/50-automation/0718-add-retry-scheduling-columns-to-webhookdeliverymodel-and-mig.md`
+Template depth: `deep`
+Reading order followed:
```
Comment (yihua-zhuo) [inline .plans/issue-718.md:13]: **INFO** (suggestion)

**Issue:** Plan cites a 'Source Contract' external dev-plan file that could not be read to verify the requirements are correctly captured.

**Suggestion:** Either inline the key requirements from the source doc or ensure the path is accessible so reviewers can cross-check intent.

File: .plans/issue-718.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            event_type="customer.created", payload={}, status="pending",
+        )
+        d = delivery.to_dict()
+        assert d["next_retry_at"] is None
+        assert d["last_attempt_at"] is None
+        assert d["error_message"] is None
+```
+
+Also update `test_columns_present` (line 68) to include the three new field names in the expected set on line 70-73.
+
```
Comment (yihua-zhuo) [inline .plans/issue-718.md:144]: **INFO** (suggestion)

**Issue:** Plan says to update test_columns_present to add the new field names, but the file already contains those fields at lines 71-73 — plan line references appear stale.

**Suggestion:** Align step descriptions with current line numbers, or remove the stale reference.
